### PR TITLE
Miscellaneous documentation updates and fixes

### LINKED ISSUE #201: Document filesystem usage and suitability for .gitignore
See https://github.com/github/gitignore/pull/1516

### PR DESCRIPTION
**Pull request by Zac-HD**
This really is a variety pack, where I tried to close a few issues and fixed other things I noticed in the process.  I suggest browsing the commit log to see the reasoning behind various changes.  Closes #201, #519, #674.

### CODE REVIEW

File: docs/changes.rst [hunk 1]
```diff
@@ -22,11 +22,18 @@ You should generally assume that an API is internal unless you have specific
 information to the contrary.
 
 -------------------
+3.11.2 - 2017-06-08
+-------------------
+
+This is a documentation release.  Several utdated snippets have been updated
```
Comment (alexwlchan) [inline docs/changes.rst]: /s/utdated/outdated

File: docs/data.rst [hunk 2]
```diff
@@ -36,9 +36,11 @@ Examples on how to use them both are below. First up choice:
         t=choice(('ST', 'LT', 'TG', 'CT'))
         # asserts go here.
 
-This means t will randomly be one of the items in the list ('ST', 'LT', 'TG', 'CT'). just like if you were calling random.choice() on the list.
+This means t will randomly be one of the items in the list ('ST', 'LT', 'TG', 'CT'),
```
Comment (alexwlchan) [inline docs/data.rst]: /s/t/it

Comment (Zac-HD) [inline docs/data.rst]: actually /s/t/`t`, but worth fixing 😄

File: docs/database.rst [hunk 3]
```diff
@@ -46,15 +46,21 @@ that generates an argument sometimes gives you data from the old strategy.
...
-problem if you just add the directory into version control.
-
-Note: There are other files in .hypothesis but everything other than the examples will be
-transparently created on demand. You don't need to and probably shouldn't check those into git.
-Adding .hypothesis/eval_source to your .gitignore or equivalent is probably a good idea.
+.. note::
+    If specific examples are important for correctness you should use the
+    :func:`@example <hypothesis.example>` decorator, as the example database may discard entries due to
+    changes in your code or dependencies.  For most users, we therefore
+    recommend using the example database as an aid for local development only.
```
Comment (alexwlchan) [inline docs/database.rst]: Mmm. Do we agree with this? Quite useful to cache it in CI as well. Do we want to add something here about caching your examples database between CI runs?

File: docs/details.rst [hunk 4]
```diff
@@ -32,12 +32,13 @@ intermediate steps of your test. That's where the ``note`` function comes in:
     ...     note("Shuffle: %r" % (ls2))
     ...     assert ls == ls2
     ...
-    >>> test_shuffle_is_noop()
+    >>> try:
+    ...     test_shuffle_is_noop()
+    ... except:
```
Comment (alexwlchan) [inline docs/details.rst]: even in examples, a bare `except` is a sad thing

File: src/hypothesis/core.py [hunk 5]
```diff
@@ -662,6 +660,8 @@ def wrapped_test(*arguments, **kwargs):
 
 
 def find(specifier, condition, settings=None, random=None, database_key=None):
+    """Returns the minimal example from the given strategy ``specifier`` that
+    matches the predicate function ``condition``."""
```
Comment (alexwlchan) [inline src/hypothesis/core.py]: tiny nit: convention elsewhere in the project is that closing triple quote is on a separate line

Comment (Zac-HD) [inline src/hypothesis/core.py]: I also think it looks weird, but `pyformat` prefers same-line quotes for a single sentence.

**Pull request merged**